worksheet_A/TerminalHacking.cpp

```diff
@@ -1,46 +1,152 @@
-// PartA_TerminalHacking.cpp : Defines the entry point for the console application.
-//
-
-#include "stdafx.h"
-#include "WordList.h"
-
-const int wordLength = 5;
-const int numberOfWords = 15;
-
-int main()
-{
-	// Seed the random number generator with the current time,
-	// to ensure different results each time the program is run
-	srand(static_cast<unsigned int>(time(nullptr)));
-
-	// Initialise word list
-	WordList words(wordLength);
-
-	// Choose secret word
-	std::string secret = words.getRandomWord();
-
-	// Create a set to hold the list of options
-	std::set<std::string> options;
-
-	// Put the secret word in the set
-	options.insert(secret);
-
-	// Fill the set with more words
-	// Using a set for options guarantees that the elements are all different
-	while (options.size() < numberOfWords)
-	{
-		std::string word = words.getRandomWord();
-		options.insert(word);
-	}
-
-	// Display all words
-	for each (std::string word in options)
-	{
-		std::cout << word << std::endl;
-	}
-
-	// TODO: implement the rest of the game
-
-    return 0;
-}
-
+// PartA_TerminalHacking.cpp : Defines the entry point for the console application.
+//
+
+#include "stdafx.h"
+#include "WordList.h"
+
+const int wordLength = 5;
+const int numberOfWords = 15;
+
+// Returns likness, the number of matching letters
+int getLikeness(std::string guessededWord, std::string secretWord)
+{
+	int result = 0;
+
+	for (int i = 0; i <= (secretWord.length() - 1); i ++)
+	{
+		if (secretWord.at(i) == guessededWord.at(i)) 
+		{
+			result += 1;
+		}
+	}
+
+	return result;
+}
+
+// Checks if word is in the list
+bool wordInList(std::string guessed, std::set<std::string> options)

```

Comment:
`std::set` has a built-in method for doing this; see http://stackoverflow.com/questions/1701067/how-to-check-that-an-element-is-in-a-stdset. Note also that the built-in method is O(log n) whereas yours is O(n), so the built-in one would be faster for larger lists of words.